tree/30_generic_methods/logger_rudder.cf

```diff
@@ -68,25 +92,72 @@ bundle agent logger_rudder(message, class_prefix)
       # 3/ Write the final expected reports, expanded. First, empty it, then fill it
       # (can't use edit_default empty, as we are iterating over a list)
     logger_rudder_temp_resfile_repaired.!logger_rudder_final_resfile_repaired.keys_defined::
-     "${expected_reports_file}"
+      "${expected_reports_file}"
         create => "true",
         edit_line => delete_lines_matching(".*");
 
+    logger_rudder_temp_resfile_repaired.!logger_rudder_final_resfile_repaired.keys_defined::
       "${expected_reports_file}"
         create    => "true",
-        edit_line => insert_lines("${reports[${keys}][0]};;${reports[${keys}][canon]};;${reports[${keys}][2]};;${reports[${keys}][3]};;${reports[${keys}][4]}"),
+        edit_line => insert_lines("${reports[${keys}][0]};;${reports[${keys}][canon]};;${reports[${keys}][2]};;${reports[${keys}][3]};;${reports[${keys}][4]};;${reports[${keys}][5]};;${reports[${keys}][6]}"),
         classes   => classes_generic("logger_rudder_final_resfile");
 
   methods:
     # no need to avoid normal ordering since this class is set when report is ready
-    report_data_read::
+    # Old reporting method
+    !report_v2.report_data_read.!no_old_class_prefix::
 
       # 4/ Array is ready, reporting time !!!
       "any" usebundle => _rudder_common_reports_generic("${report_data[1]}", "${c_class_prefix}", "${report_data[3]}", "${report_data[4]}", "${report_data[5]}", "${message}"),
         classes    => classes_generic("logger_rudder_${c_class_prefix}");
 
+    # We HAVE the data and we CAN dot it -> New reporting method
+    report_v2::
```

Comment:
You should at least have logger_rudder_final_resfile no ?
Otherwise the file will not have been finished; or is it is not necessary, then in case of v2 report, the file .res should not be created